tests/python/test_type_promotion.py

```diff
@@ -60,3 +60,17 @@ def func():
     func()
 
     assert np.allclose(y.to_numpy(), np.sqrt(x.to_numpy()))
+
+
+@test_utils.test()
+def test_shift_ops():
```

Comment:
I suggest writing cpp tests for type check. FYI https://github.com/taichi-dev/taichi/blob/master/tests/cpp/ir/frontend_type_inference_test.cpp

Reply:
sounds good. Let me see if I can write a similar one at CHI IR level.

Reply:
Switched to Cpp test case. Thanks!